lib/storage/src/content_manager/toc/collection_meta_ops.rs

```diff
@@ -191,6 +191,14 @@ impl TableOfContent {
             });
             Ok(true)
         } else {
+            let path = self.get_collection_path(collection_name);
+            if path.exists() {
+                log::warn!(
+                    "Collection {} is not loaded, but its directory still exists. Deleting it.",
+                    collection_name
+                );
+                tokio::fs::remove_dir_all(path).await?;
+            }
```

Comment:
Are we 100% sure this won't delete by accident? That would be painful.

Can we possibly improve it by checking if any locks are held inside the collection directory?

I can't think of a situation where this happens, but I'd prefer to be as careful as possible here.

Reply:
well, collection is explicitly requested to be deleted. If it existed - it would have been deleted anyway.

I am not sure about locks and how can it potentially help

Reply:
I misread. I thought this was in the collection creation call, as a mechanism to help with collection creation.

I see now. No objections. Thanks!